### ISSUE TITLE
Automatically add  --enable-preview to vmArgs, when necessary

### ISSUE THREAD

**Issue by fbricon** [Microsoft/vscode-java-debug#553]
Following on https://github.com/redhat-developer/vscode-java/issues/671, when trying to run Java 12 code using preview features, you need to add the `--enable-preview` flag in the launch configuration's `vmArgs`:
```json
{
    "configurations": [
        {
            "type": "java",
            "name": "CodeLens (Launch) - Demo",
            "request": "launch",
            "mainClass": "foo.bar.Demo",
            "projectName": "Demo",
            "vmArgs": "--enable-preview"
        }
    ]
}
```

This could be done automatically by the vscode-java-debug extension

**Comment by testforstephen** [comment]
This `vmArgs` is not recognized by Java 8. We need distinguish the JVM version first and determine whether to add this flag.

**Comment by testforstephen** [comment]
The fix is detecting the java project settings firstly. if the debugger found the current java project enabled preview compiler flag, then auto add `--enable-preview` to vmArgs.

In a word, you don't need manually specify `--enable-preview` vmArgs in launch.json any more.

**Comment by szx233** [comment]
> The fix is detecting the java project settings firstly. if the debugger found the current java project enabled preview compiler flag, then auto add `--enable-preview` to vmArgs.
> 
> In a word, you don't need manually specify `--enable-preview` vmArgs in launch.json any more.

hi,have you update this? I still have this problem,it only works after adding the 'vmArgs'.

**Comment by testforstephen** [comment]
@szx233 We just released 0.18.0, which should have fixed this issue.
